allennlp_models/vision/dataset_readers/flickr30k.py

```diff
@@ -88,6 +90,12 @@ class Flickr30kReader(VisionReader):
         the captions and metadata for each task instance.
     tokenizer: `Tokenizer`, optional
     token_indexers: `Dict[str, TokenIndexer]`
+    featurize_captions: `bool`, optional
+        If we should featurize captions while calculating hard negatives, or use placeholder features.
+    is_evaluation: `bool`, optional
+        If the reader should return instances for evaluation or training.
+    n: int, optional
+        The number of potential hard negatives to consider.
```

Comment:
Call it `num_hard_negatives` then?

Reply:
Has that always been configurable? I thought we were always using one hard negative?

Reply:
This is a different variable, that I guess doesn't need to be configurable. We first find the `n` (default = `100`) closest images based on the L2 distance between their averaged features, then out of those `n`, we do the actual calculations with the caption vectors. I'm fine with removing it or renaming it, what do you think?